### PR TITLE
Issues/153

### PR DESCRIPTION
**Pull request by noahl**
Support ssh key files with passphrases

This ended up being some relatively small changes to how we use pexpect, but figuring out which small changes we needed took a long time.

One thing that is not in this PR (yet) is the changes to `Vagrantfile` that I used to test this, because I haven't figured out how to make them portable yet. I'm not sure if they need to be in this PR, but I do want to merge them eventually.

### CODE REVIEW

File: doc/command_syntax_usage.rst [hunk 1]
```diff
@@ -38,6 +38,20 @@ over SSH. Each authentication identity requires its own auth entry.
 
 *Note:* --password not being passed or passed as empty are considered the same thing.
 
+---------------------------
+Password-protected SSH key
```
Comment (chambridge) [inline doc/command_syntax_usage.rst]: I've heard this referred to more often as **SSH Key with a Passphrase**. But I'm flexible if you have heard it differently.

**Review by chambridge — APPROVED** [review]
Minor doc comment.

File: doc/command_syntax_usage.rst (hunk 1 above)
Comment (noahl) [reply doc/command_syntax_usage.rst]: Changed to your title; thank you!

File: doc/command_syntax_usage.rst [hunk 2]
```diff
@@ -38,9 +38,9 @@ over SSH. Each authentication identity requires its own auth entry.
...
 *Note:* --password not being passed or passed as empty are considered the same thing.
 
----------------------------
-Password-protected SSH key
----------------------------
+-------------------------
+SSH Key with a Passphrase
+-------------------------
 
 There are two ways to handle password-protected ssh keys. If a
```
Comment (chambridge) [inline doc/command_syntax_usage.rst]: Sorry to nit pick but I would also think we would use the same language in the paragraph as well as the change to the section name.

Comment (noahl) [reply doc/command_syntax_usage.rst]: Sorry, just saw this comment after merging. I will fix the paragraph too.

**Pull request merged**